helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/PartitionMovementConstraint.java

```diff
@@ -54,23 +52,18 @@
   @Override
   protected float getAssignmentScore(AssignableNode node, AssignableReplica replica,
       ClusterContext clusterContext) {
+    // Prioritize the matching of the previous Best Possible assignment.
     Map<String, String> bestPossibleStateMap =
         getStateMap(replica, clusterContext.getBestPossibleAssignment());
+    if (!bestPossibleStateMap.isEmpty()) {
+      return calculateAssignmentScale(node, replica, bestPossibleStateMap);
+    }
     Map<String, String> baselineStateMap =
         getStateMap(replica, clusterContext.getBaselineAssignment());
-
-    // Prioritize the matching of the previous Best Possible assignment.
-    float scale = calculateAssignmentScale(node, replica, bestPossibleStateMap);
-    // If the baseline is also provided, adjust the final score accordingly.
-    scale = scale * (1 - BASELINE_MATCH_FACTOR)
-        + calculateAssignmentScale(node, replica, baselineStateMap) * BASELINE_MATCH_FACTOR;
-
-    return scale;
-  }
-
-  @Override
-  NormalizeFunction getNormalizeFunction() {
-    return score -> score * (getMaxScore() - getMinScore()) + getMinScore();
+    if (!baselineStateMap.isEmpty()) {
+      return calculateAssignmentScale(node, replica, baselineStateMap);
+    }
+    return 0;
```

Comment:
How come we don't use a "smoothing" function here?

Reply:
Tried, failed, no better idea for this constraint.
You are welcome to nominate some candidates : )